tests/src/test/scala/it/unibo/scafi/test/functional/stdlib/BlockT.scala

```diff
@@ -0,0 +1,117 @@
+package it.unibo.scafi.test.functional.stdlib
+
+import it.unibo.scafi.test.FunctionalTestIncarnation._
+import it.unibo.scafi.test.functional.ScafiAssertions.assertNetworkValues
+import it.unibo.scafi.test.functional.ScafiTestUtils
+import it.unibo.utils.StatisticsUtils._
+import org.scalatest._
+
+
+/*
+Not yet tested:
+- cyclicTimerWithDecay
+- clock
+- impulsesEvery
+ */
+class BlockT extends FlatSpec{
+  import ScafiTestUtils._
+
+  val Block_T = new ItWord
+
+  private[this] trait SimulationContextFixture {
+    val net: Network with SimulatorOps = ScafiTestUtils.standardNetwork()
+  }
+
+  private[this] trait TestProgram extends AggregateProgram with StandardSensors with BuildingBlocks
+
+  def unitaryDecay: Int => Int = _ - 1
+  def halving: Int => Int = _ / 2
+
+  Block_T should "support T with unitary decay and 0 floor value" in new SimulationContextFixture {
+    exec(new TestProgram {
+      override def main(): Any = T(10, 0, unitaryDecay)
+    }, ntimes = someRounds)(net)
+
```

Comment:
We should also test that the devices compute 10 initially, and possibly intermediate states as well. In order to do this, we may e.g. manually regulate the scheduling and keep track of the number of rounds a device has computed